scaladoc-js/src/searchbar/code-snippets/CodeSnippets.scala

```diff
@@ -0,0 +1,37 @@
+package dotty.tools.scaladoc
+
+import org.scalajs.dom._
+import org.scalajs.dom.ext._
+
+class CodeSnippets:
+  val replacePatternsAndResults: Seq[(String, String)] = Seq(
+    """(\/\/{{\n)((.|\n)*?)(\/\/}}\n)""" -> """<span class="hideable">$2</span>""", // wrap content of block directives
+    """(\/\/.*?)(\n|\$)""" -> """<span class="hideable">$1</span>$2""", // wrap single line comment
+    """(\/\*)((.|\n)*?)(\*\/)""" -> """<span class="hideable">$0</span>""", // wrap multi line comment
+  )
+
+  document.querySelectorAll("code").foreach {
+    case e: html.Element => e.innerHTML = replacePatternsAndResults.foldLeft(e.innerHTML) {
+      case (acc, (pattern, result)) =>
+        acc.replaceAll(pattern, result)
+    }
+  }
+
+  def toggleHide(e: html.Element | html.Document) = e.querySelectorAll("code span.hideable").foreach {
+    case e: html.Element if e.style.getPropertyValue("display").isEmpty => e.style.setProperty("display", "none")
+    case e: html.Element => e.style.removeProperty("display")
+  }
+
+  toggleHide(document)
+
+  document.querySelectorAll("pre").foreach {
+    case e: html.Element =>
+      val a = document.createElement("a")
+      a.textContent = "Show"
+      a.addEventListener("click", { (_: MouseEvent) =>
+        a.textContent = if a.textContent == "Show" then "Hide" else "Show"
```

Comment:
I think it might be misleading: just "Show" and just "Hide" may indicate that it applies to the whole snippet. I like the way it was done in Dokka - there was "+" and "-" which enlarged and reduced snippet. Maybe you will invent sth better

Reply:
You're right about misleading texts. If we want to stick to the word names rather than some symbols, maybe `Expand` and `Contract`. Anyway we could think of some more modern design.

Reply:
For reference, here is how dokka handles it
![obraz](https://user-images.githubusercontent.com/32793002/112126090-64725f00-8bc4-11eb-9162-d54d87e60ef5.png)

![obraz](https://user-images.githubusercontent.com/32793002/112126134-6f2cf400-8bc4-11eb-9b6a-1f0941dd1978.png)